src/snowflake/snowpark/mock/analyzer.py

```diff
@@ -288,12 +289,12 @@ def analyze(
             return expr.sql
 
         if isinstance(expr, Literal):
-            sql = str(expr.value)
+            sql = to_sql(expr.value, expr.datatype)
             if parse_local_name:
                 sql = sql.upper()
             # single quote literal when called by function/method to describe the column name of a dataframe
             # this is to align with the behavior of snowpark python when running against snowflake.
-            return f"'{sql}'"
+            return f"{sql}"
```

Comment:
I removed this to pass `test_cast_nan_column_name` but the change seems to not break any tests. Is this expected? @sfc-gh-aling

Reply:
For my education, what was the purpose of adding the single quotes before?

Reply:
this is to add single quote around a literal, e.g., `select colname where colname like 'word'`
I think `to_sql` (which calls `str_to_sql`) handles the literal case.